src/BuildIt.Forms/BuildIt.Forms.Controls/CameraPreviewControl.xaml.cs

```diff
@@ -106,6 +153,42 @@ public Aspect Aspect
         /// </summary>
         internal Func<Task<IReadOnlyList<ICamera>>> RetrieveCamerasFunc { get; set; }
 
+        /// <summary>
+        /// Start camera preview
+        /// </summary>
+        /// <returns></returns>
+        public async Task StartPreview()
+        {
+            if (StartPreviewFunc == null || Status == CameraStatus.Started || Status == CameraStatus.Starting)
+            {
+                return;
+            }
+
+            if (await startCameraPreviewSemaphoreSlim.WaitAsync(0))
```

Comment:
I wasn't sure what we want to do here, if we want to just gracefully return or we want to actually notify the caller about Start already being in progress?